src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts

```diff
@@ -0,0 +1,1275 @@
+import { useDispatch, useSelector } from "react-redux";
+import {useCallback, useContext} from 'react';
+import { TreeItem } from "react-complex-tree";
+
+import { 
+	MainContext,
+	expandFFNode, 
+	ffSelector, 
+	navigatorSelector, 
+	setCurrentFile, 
+	setFileAction, 
+	updateFFTreeViewState 
+} from "@_redux/main";
+import { TFileNodeData, createDirectory, getStat, removeFileSystem, writeFile } from "@_node/file";
+import { TNode, TNodeTreeData, TNodeUid, TNormalNodeData } from "@_node/types";
+import { TFileAction, TFileNodeType } from "@_types/main";
+import { verifyFileHandlerPermission } from "@_services/main";
+import { 
+	deletingWarning,
+	duplicatingWarning,
+	fileError, 
+	folderError, 
+	invalidDirError, 
+	movingError, 
+	renamingError 
+} from "../errors";
+import { useReloadProject } from "./useReloadProject";
+import { HmsClearActionType, RootNodeUid, TmpNodeUid } from "@_constants/main";
+import { useInvalidNodes } from "./useInvalidNodes";
+import { useTemporaryNodes } from "./useTemporaryNodes";
+import { getValidNodeUids } from "@_node/apis";
+import { moveActions } from "../helpers";
+
+export const useNodeActionsHandler = (openFileUid: React.MutableRefObject<string>) =>{
+
+  const dispatch = useDispatch();
+  const { file } = useSelector(navigatorSelector);
+
+  const {
+    // global action
+    addRunningActions,
+    removeRunningActions,
+    // navigator
+    project,
+    setNavigatorDropDownType,
+    // file tree view
+    ffTree,
+    setFFTree,
+    ffHandlers,
+    setCurrentFileUid,
+    // code view
+    showCodeView,
+    setShowCodeView,
+    // processor
+    setUpdateOpt,
+    // references
+    htmlReferenceData,
+    // cmdk
+    // other
+    osType,
+    // toasts
+    addMessage,
+    // non-parse file
+    setParseFile,
+    setPrevFileUid,
+  } = useContext(MainContext);
+
+  const {
+    focusedItem,
+    expandedItemsObj,
+    selectedItems,
+  } = useSelector(ffSelector);
+
+  const { cb_reloadProject } = useReloadProject();
+
+  const { removeInvalidNodes, setInvalidNodes, invalidNodes } = useInvalidNodes();
+
+  const { setTemporaryNodes, removeTemporaryNodes } = useTemporaryNodes();
+
+  const { moveIDBFF, moveLocalFF } = moveActions();
+	
+	const createFFNode = useCallback(
```

Comment:
Break all the if else blocks with large number of locs into smaller functions and then call them inside those blocks. You can use the concept of pure function if there is only transformation of input to an output is there.

Function names should be proper